dask_cuda/dgx.py

```diff
@@ -1,116 +1,102 @@
+        Raises
+        ------
+        TypeError
+            If enable_infiniband or enable_nvlink is True and protocol is not 'ucx'
 
-    Examples
-    --------
-    >>> from dask_cuda import DGX
-    >>> from dask.distributed import Client
-    >>> cluster = DGX()
-    >>> client = Client(cluster)
-    """
-    if (
-        enable_tcp_over_ucx or enable_infiniband or enable_nvlink
-    ) and protocol != "ucx":
-        raise TypeError("Enabling InfiniBand or NVLink requires protocol='ucx'")
+        Examples
+        --------
+        >>> from dask_cuda import DGX
+        >>> from dask.distributed import Client
+        >>> cluster = DGX()
+        >>> client = Client(cluster)
+        """
+        if (
+            enable_tcp_over_ucx or enable_infiniband or enable_nvlink
+        ) and protocol != "ucx":
+            raise TypeError("Enabling InfiniBand or NVLink requires protocol='ucx'")
 
-    ucx_net_devices = ""
-    if enable_infiniband:
-        ucx_net_devices = lambda i: "mlx5_%d:1" % (i // 2)
+        initialize(
+            enable_tcp_over_ucx=enable_tcp_over_ucx,
+            enable_infiniband=enable_infiniband,
+            enable_nvlink=enable_nvlink,
+        )
 
-    spec = worker_spec(
-        interface=interface,
-        dashboard_address=dashboard_address,
-        threads_per_worker=threads_per_worker,
-        silence_logs=silence_logs,
-        CUDA_VISIBLE_DEVICES=CUDA_VISIBLE_DEVICES,
-        enable_tcp_over_ucx=enable_tcp_over_ucx,
-        enable_infiniband=enable_infiniband,
-        ucx_net_devices=ucx_net_devices,
-        enable_nvlink=enable_nvlink,
-        protocol=protocol,
-        **kwargs,
-    )
+        self.set_ucx_net_devices = enable_infiniband
 
-    scheduler = {
-        "cls": Scheduler,
-        "options": {
-            "interface": interface,
-            "protocol": protocol,
-            "dashboard_address": dashboard_address,
-            **get_preload_options(
-                protocol=protocol,
-                enable_tcp_over_ucx=enable_tcp_over_ucx,
-                enable_infiniband=enable_infiniband,
-                enable_nvlink=enable_nvlink,
-            ),
-        },
-    }
+        super().__init__(
+            interface=interface,
+            dashboard_address=dashboard_address,
+            threads_per_worker=threads_per_worker,
+            silence_logs=silence_logs,
+            CUDA_VISIBLE_DEVICES=CUDA_VISIBLE_DEVICES,
+            protocol=protocol,
+            **kwargs,
+        )
 
-    return SpecCluster(
-        workers=spec, scheduler=scheduler, silence_logs=silence_logs, **kwargs
-    )
+    def new_worker_spec(self):
+        spec = super().new_worker_spec()
+        first_spec = next(iter(spec.values()))
+        dev = int(first_spec["options"]["env"]["CUDA_VISIBLE_DEVICES"].split(",")[0])
+        if self.set_ucx_net_devices:
+            first_spec["options"]["env"]["UCX_NET_DEVICES"] = "mlx5_%d:1" % (dev // 2)
+        return spec
```

Comment:
I think you're not configuring CPU affinity for workers here, that is required.

Reply:
Notice, I use `spec = super().new_worker_spec()` to call [`LocalCUDACluster.new_worker_spec()` ](https://github.com/rapidsai/dask-cuda/blob/953b21bb6a4226fdf5d55989568851863430490c/dask_cuda/local_cuda_cluster.py#L112). 
I think that configures the CPU affinity or  am I missing something?

Reply:
Ah I see, I missed that you subclassed `LocalCUDACluster`. Thanks for pointing it out!